Populate user_data with snippet metadata

alindeman opened the pull request:
If the langserver returns a snippet ([`insertTextFormat == 2`](https://microsoft.github.io/language-server-protocol/specification#textDocument_completion)), populating the `user_data` in vim/neovim will allow snippet managers like [neosnippet](https://github.com/Shougo/neosnippet.vim/blob/f7755084699db69ce9ff51c87baf8e639b7e543a/autoload/neosnippet/mappings.vim#L166-L175) to trigger snippet completion.

[![demo](https://asciinema.org/a/yyHjExpQyu5piYI8TY9uxf7jL.svg)](https://asciinema.org/a/yyHjExpQyu5piYI8TY9uxf7jL)

```viml
let g:neosnippet#enable_completed_snippet = 1
autocmd CompleteDone * call neosnippet#complete_done()
```

natebosch commented:
Looks interesting! This might be enough to consider #82 solved.

Did you separately use `message_hooks` or something to set the client capability for snippets to enable this? Or is the server you're using not spec compliant and sending snippets even though the client reports it can't use them?

I'm wondering if we need a global variable to set to update the client capabilities for this.

alindeman commented:
@natebosch Good catch. I discovered `vim-lsc` was sending `snippetSupport` under the wrong key. I fixed that and introduced a configuration option in the latest commit pushed to this branch. Let me know if that makes sense or you have additional ideas on tweaks to make.

natebosch commented on doc/lsc.txt:
```diff
@@ -315,6 +315,14 @@ be smaller than with full syncs.
+By default the client disables support for snippets as insert text. If set to
+`v:true` and supported by the server, insert text may contain snippets.
+
+Autocompletion of snippets requires a snippet plugin such as ultisnips or
```
I think we should expand this. Something like "Autocompletion of snippets requires a snippet plugin which can complete using `user_data` on `CompleteDone`, such as neosnippet.vim"
